docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -726,6 +726,58 @@ const classes = makeStyles(theme => ({
   +<Slider onChange={(event: React.SyntheticEvent, value: unknown) => {}} />
   ```
 
+- The `ValueLabelComponent` prop is now part of the `components` prop.
+
+  ```diff
+  -<Slider ValueLabelComponent={CustomValueLabel} />
+  +<Slider components={{ ValueLabel: CustomValueLabel }} />
+  ```
+
+- The `ThumbComponent` prop is not part of the `components` prop.
+
+  ```diff
+  -<Slider ThumbComponent={CustomThumb} />
+  +<Slider components={{ Thumb: CustomThumb }} />
+  ```
+
+- The `classes` prop is no longer available for styling the component. You can fix this by one of these options:
```

Comment:
Agree, in the end we may need to come back to support this API again, but let's see the feedback first

Reply:
I think we can save the community discussion, and agree that this will be a breaking change too far, with potential repetitional consequences. It doesn't need to be resolved in this PR, but it needs to be resolved.

Reply:
We will need to create separate component for each parts of the slider, with that we can offer support for this. As far as I remember we had a PR with this changes and I can reopen it. There was some perf implication, but I will reopen the PR tomorrow and share more info.

Reply:
I don’t think it makes sense to merge this one if we all agree that we should suppor the classes prop, I think we need to add that support first

Reply:
Let's wait for feedback from other in @mui-org/core-team.

Reply:
I think I voiced this concern already. Unless we have a robust codemod that automates migrating away from `classes` (and I don't think this is feasible), migrating might take a very long time. And without `classes` we're also loosing quite a bit of (literal) type-safety. I was always under the impression that this prop stays anyway.

Reply:
Here is a PR - https://github.com/mui-org/material-ui/pull/22893 should we first do a review on that one?

Reply:
I've marked this one as on-hold.

Reply:
Regarding the pros & cons, we started to discuss it in https://github.com/mui-org/material-ui/issues/22342#issuecomment-703711215.